app/src/main/java/com/pavesid/androidacademy/retrofit/CacheControlInterceptor.kt

```diff
@@ -0,0 +1,22 @@
+package com.pavesid.androidacademy.retrofit
+
+import com.pavesid.androidacademy.utils.Utils
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class CacheControlInterceptor : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val originalResponse = chain.proceed(chain.request())
+        return if (Utils.isNetworkConnected) {
+            val maxAge = 60 // read from cache for 1 minute
+            originalResponse.newBuilder()
+                .header("Cache-Control", "public, max-age=$maxAge")
+                .build()
```

Comment:
I'm not sure that it's work. At least, when I load the first page of movies, then click the back button and open the app after 5 minutes - in Profiler at Network tab shows nothing.

Reply:
I'm not sure either, but in theory it should.